db/crud.go

```diff
@@ -886,25 +860,12 @@ func (db *Database) PutExistingRev(newDoc *Document, docHistory []string, noConf
 	return doc, newRev, err
 }
 
-func (db *Database) PutExistingRevWithBody(docid string, body Body, docHistory []string, noConflicts bool) (doc *Document, newRev string, err error) {
-	expiry, _ := body.ExtractExpiry()
-	deleted := body.ExtractDeleted()
-	revid := body.ExtractRev()
-
-	newDoc := &Document{
-		ID:        docid,
-		Deleted:   deleted,
-		DocExpiry: expiry,
-		RevID:     revid,
+func (db *Database) PutExistingRevWithBody(docid string, body Body, docHistory []string, noConflicts bool) (doc *Document, newRevID string, err error) {
+	newDoc, err := body.ToIncomingDocument()
```

Comment:
docHistory is being generated from the equivalent of SpecialProperties.revisions (in handlePutDoc, handleBulkDocs).  The fact that it's not being set on the IncomingDocument here suggests that IncomingDocument isn't being created at the ideal point in the stack.